Assets/Scripts/EnemyAI.cs

```diff
@@ -158,310 +164,315 @@ void Start()
 
     private void FixedUpdate()
     {
-        //Checks
-        RaycastHit seePlayer;
-        Ray lookForPlayer = new Ray(transform.position, toPlayer);
-        Physics.Raycast(lookForPlayer, out seePlayer, terrainPlayerLayer);
-        if (seePlayer.transform != null)
+        //Calculating sight range
+        toPlayer = player.transform.position - transform.position;
+
+        //checking if player is within sight range
+        Ray lookForPlayer = new Ray(transform.position + Vector3.up*0.25f, toPlayer); //added Vector3.up the origin of ray for better collisions
+        if (Physics.Raycast(lookForPlayer, out RaycastHit hit, terrainPlayerLayer))
         {
-            playerInSightRange = (seePlayer.distance < sightRange && seePlayer.transform.gameObject.name == player.transform.GetChild(1).name);
+            playerInSightRange = (hit.distance < sightRange && hit.transform.gameObject.name == player.transform.GetChild(1).name);
```

Comment:
Delete extra capsule, use maxDistance parameter and, use compareTag.
optional simplification:
`playerInSightRange = Physics.Raycast(...out RaycastHit hit...) && hit.compareTag("Player")`